code/game/gamemodes/cult/ritual.dm

```diff
@@ -82,762 +88,103 @@ var/engwords = list("travel", "blood", "join", "hell", "destroy", "technology",
-				user.take_overall_damage((rand(9)+1)/10) // 0.1 to 1.0 damage
-				if(do_after(user, 50, target = user))
-					if(usr.get_active_hand() != src)
-						return
-					var/mob/living/carbon/human/H = user
-					var/obj/effect/rune/R = new /obj/effect/rune(user.loc)
-					user << "<span class='danger'>You finish drawing the arcane markings of the Geometer.</span>"
-					var/list/required = dictionary[chosen_rune]
-					R.word1 = english[required[1]]
-					R.word2 = english[required[2]]
-					R.word3 = english[required[3]]
-					R.check_icon()
-					R.blood_DNA = list()
-					R.blood_DNA[H.dna.unique_enzymes] = H.dna.blood_type
-				return
-	else
-		user << "The book seems full of illegible scribbles. Is this a joke?"
-		return
-
-/obj/item/weapon/tome/attackby(obj/item/weapon/tome/T, mob/living/user, params)
-	if(istype(T, /obj/item/weapon/tome)) // sanity check to prevent a runtime error
-		switch(alert("Copy the runes from your tome?",,"Copy", "Cancel"))
-			if("cancel")
-				return
-	//	var/list/nearby = viewers(1,src) //- Fuck this as well. No clue why this doesnt work. -K0000
-	//		if (T.loc != user)
-	//			return
-	//	for(var/mob/M in nearby)
-	//		if(M == user)
-		for(var/entry in words)
-			words[entry] = T.words[entry]
-		user << "<span class='notice'>You copy the translation notes from your tome.</span>"
-
-
-/obj/item/weapon/tome/examine(mob/user)
-	..()
-	if(iscultist(user))
-		user << "The scriptures of Nar-Sie, The One Who Sees, The Geometer of Blood. Contains the details of every ritual his followers could think of."
-
-/obj/item/weapon/tome/imbued //admin tome, spawns working runes without waiting
-	w_class = 2.0
-	var/cultistsonly = 1
-	attack_self(mob/user as mob)
-		if(src.cultistsonly && !iscultist(usr))
+			cultist_commune(user, 1, 0, input)
+
+/obj/item/weapon/tome/proc/read_tome(mob/user)
+	user << "<span class='warning'>You can't seem to understand the tome's scribbles. Perhaps this will become better revealed, in time?</span>"
+
+/obj/item/weapon/tome/proc/scribe_rune(mob/user)
+	var/rune_to_scribe = input(user, "Choose a rune to scribe.", "Sigils of Power") as null|anything in allowed_runes
+	user.visible_message("<span class='warning'>[user] cuts open their arm and begins writing in their own blood!</span>", \
+						 "<span class='danger'>You slice open your arm and begin drawing a sigil of the Geometer.</span>")
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		C.apply_damage(1, BRUTE, pick("l_arm", "r_arm"))
+	if(!do_after(user, 50, target = get_turf(user)))
+		return
+	user.visible_message("<span class='warning'>[user] creates a strange circle in their own blood.</span>", \
+						 "<span class='danger'>You finish drawing the arcane markings of the Geometer.</span>")
+	var/obj/effect/rune/rune_type
+	switch(rune_to_scribe)

```

Comment:
I'd look into the ```typesof``` proc so you can dynamically get all runes, unless you don't want that.
(Also, if you do decide to do this, I'd cache the result because you can't add more types at runtime.)